jwt/api_jws.py

```diff
@@ -69,8 +74,13 @@ def get_algorithms(self):
         """
         return list(self._valid_algs)
 
-    def encode(self, payload, key, algorithm='HS256', headers=None,
-               json_encoder=None):
+    def encode_bytes(self,
+                     payload,  # type: bytes
```

Comment:
I'm wondering if we could leave this as `.encode()` and make the type for payload `Union[bytes, Dict]`?

Reply:
It's possible, doing so mypy says:
 `jwt/api_jwt.py:57: error: Unsupported target for indexed assignment`
because the function is using the payload it as a dictionary. But before that there's a check for it and an error being thrown if it's not so I think it's fine to silence it.

The new commit does that and revert `encode_bytes` to `encode`